purchases/src/test/java/com/revenuecat/purchases/AppConfigTest.kt

```diff
@@ -0,0 +1,214 @@
+        val appConfig = AppConfig(
+            context = mockContext,
+            observerMode = false,
+            platformInfo = PlatformInfo(flavor = "native", version = "3.2.0"),
+            proxyURL = null
+        )
+        assertThat(appConfig.languageTag).isEqualTo(expected)
+    }
+
+    @Test
+    fun `languageTag defaults to empty string`() {
+        val expected = ""
+        val mockContext = mockk<Context>(relaxed = true)
+        mockkStatic("com.revenuecat.purchases.UtilsKt")
+        every {
+            mockContext.getLocale()?.toBCP47()
+        } returns null
+        val appConfig = AppConfig(
+            context = mockContext,
+            observerMode = false,
+            platformInfo = PlatformInfo(flavor = "native", version = "3.2.0"),
+            proxyURL = null
+        )
+        assertThat(appConfig.languageTag).isEqualTo(expected)
+    }
+
+    @Test
+    fun `versionName is created successfully`() {
+        val expected = "1.0.0"
+        mockkStatic("com.revenuecat.purchases.UtilsKt")
+        val mockContext = mockk<Context>(relaxed = true) {
+            every {
+                versionName
+            } returns expected
+        }
+
+        val appConfig = AppConfig(
+            context = mockContext,
+            observerMode = false,
+            platformInfo = PlatformInfo(flavor = "native", version = "3.2.0"),
+            proxyURL = null
+        )
+        assertThat(appConfig.versionName).isEqualTo(expected)
+    }
+
+    @Test
+    fun `versionName defaults to empty string`() {
+        val expected = ""
+        mockkStatic("com.revenuecat.purchases.UtilsKt")
+        val mockContext = mockk<Context>(relaxed = true) {
+            every {
+                versionName
+            } returns null
+        }
+        val appConfig = AppConfig(
+            context = mockContext,
+            observerMode = false,
+            platformInfo = PlatformInfo(flavor = "native", version = "3.2.0"),
+            proxyURL = null
+        )
+        assertThat(appConfig.versionName).isEqualTo(expected)
+    }
+
+    @Test
+    fun `finishTransactions is set correctly when observer mode is false`() {
+        val appConfig = AppConfig(
+            context = mockk(relaxed = true),
+            observerMode = false,
+            platformInfo = PlatformInfo(flavor = "native", version = "3.2.0"),
+            proxyURL = null
+        )
+        assertThat(appConfig.finishTransactions).isTrue()
+    }
+
+    @Test
+    fun `finishTransactions is set correctly when observer mode is true`() {
+        val appConfig = AppConfig(
+            context = mockk(relaxed = true),
+            observerMode = true,
+            platformInfo = PlatformInfo(flavor = "native", version = "3.2.0"),
+            proxyURL = null
+        )
+        assertThat(appConfig.finishTransactions).isFalse()
+    }
+
+    @Test
+    fun `proxyURL is set as a baseURL`() {
+        val expected = URL("https://a-proxy")
```

Comment:
we have a slight disconnect between our proxyURL implementations for iOS / Android: 
- iOS only requires the domain for the proxy, whereas Android also requires protocol

| iOS | Android |
| :-: | :-: |
| `a-proxy` | `https://a-proxy` |

We should try to unify. 
My reasoning for going without protocol was to pretty much remove the ability for developers to use http, since it would make things less secure and it would totally defeat the purpose of the proxy for kids apps.

However, not having the protocol does make things slightly more awkward in two ways: 
- if you want to test against local khepri, you have to edit the SDK in order to switch protocols
- it's called a `proxyURL` but it's not technically a URL without the protocol. 

What are your thoughts on this? @vegaro @rkotzy

We could also solve it by asking developers to always start with `https://` (like android does) and issuing a warning if they don't. 
In either case, we should try to perform some validation to make sure that it's a valid URL and that protocol is either http or https, and crash if it isn't - that way it's super easy to catch. Also, the SDK will pretty much not work with an invalid URL.

Reply:
I used `URL` because it gives us some basic validation out of the box, plus Android would give this error if using http:

`Cleartext HTTP traffic to api.revenuecat.com not permitted`

I believe that's starting with Android 9, so not all Android devices will give that error. Can we accept a NSURL and check for the protocol in iOS? Maybe we can add the check in Android too, to get the error in all devices, not only in > Android 9

Reply:
thinking about this again, it's definitely easier to just take the URL, like here. 
on iOS, you have to explicitly add a bypass to AppTransport security to even use http in the first place, and Apple will actually even follow up asking why (I've seen it happen). 
I'll refactor 👍